### PR TITLE
exclude mpi run from auto-detection of horovod

### LINKED ISSUE #8573: SLURM mpirun/openmpi training selects horovod automatically when not requested
## 🐛 Bug

I tried to do a hyperparameter search using wandb sweep function on a SLURM cluster by mpirun (openmpi 4.0).
However, during training always horovod as distributed backend is tried to be selected though the trainer config is accelerator=None/distributed_backend=None/gpus=1.

Error:
                'Requested `accelerator="horovod"`, but Horovod is not installed.'
                "Install with \n $HOROVOD_WITH_PYTORCH=1 pip install horovod[pytorch]"

I use mpirun like this

mpirun \
    --display-map \
    --display-allocation \
    --map-by ppr:2:socket:pe=19 \
    bash -c 'export CUDA_VISIBLE_DEVICES=${OMPI_COMM_WORLD_LOCAL_RANK}; wandb agent SWEEPID'

Also tried to install $HOROVOD_WITH_PYTORCH=1 pip install horovod[pytorch]. But in this case, training won't start and hangs in a never ending loop (but no error occurs).

Using 

CUDA_VISIBLE_DEVICES=0 wandb agent $SWEEPID &
CUDA_VISIBLE_DEVICES=1 wandb agent $SWEEPID &
CUDA_VISIBLE_DEVICES=2 wandb agent $SWEEPID &
CUDA_VISIBLE_DEVICES=3 wandb agent $SWEEPID 

works fine.

### Environment
 - PyTorch Lightning Version: 1.2.10
 - PyTorch Version: 1.8
 - Python version: 3.8.0
 - OS (e.g., Linux): Linux on SLURM cluster
 - openmpi 4.0
 - CUDA/cuDNN version: 11.2
 - GPUs: one node with 4 Tesla A100
 - How you installed PyTorch (`conda`, `pip`, source): conda

Do you have any idea, why horovod is implictly selected instead of doing 4 "normal trainings in parallel" with different parameter settings? 

Thanks in advance!

Best,
Marcel

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #8573

Horovod can be launched by either horovodrun (convenience utility) or by equivalent commands through mpirun directly (see [docs](https://horovod.readthedocs.io/en/stable/mpi_include.html#mpirun)). However, since mpirun is a general purpose HPC technology we probably don't want to autodetect horovod unless horovod is actually installed. 

I'm very unfamiliar with the details of the mpirun + horovod usage, so suggestions are welcome.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
   - No, suggestions welcome. I tried to write a test but having trouble mocking the horovod imports pretending
      it is not installed. If you have any ideas, please let me know.
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
Great catch !

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
